pkg/collector/corechecks/sbom/processor.go

```diff
@@ -189,36 +189,47 @@ func (p *processor) processHostRefresh() {
 
 	go func() {
 		result := <-ch
+		log.Debugf("processing host scanresult: %v", result)
 
-		if result.Error != nil {
-			// TODO: add a retry mechanism for retryable errors
-			log.Errorf("Failed to generate SBOM for host: %s", result.Error)
-			return
-		}
-
-		log.Debugf("Successfully generated SBOM for host: %v, %v", result.CreatedAt, result.Duration)
-
-		bom, err := result.Report.ToCycloneDX()
-		if err != nil {
-			log.Errorf("Failed to extract SBOM from report: %s", err)
-			return
-		}
-
-		p.queue <- &model.SBOMEntity{
+		sbom := &model.SBOMEntity{
+			Status:             model.SBOMStatus_SUCCESS,
 			Type:               model.SBOMSourceType_HOST_FILE_SYSTEM,
 			Id:                 p.hostname,
-			GeneratedAt:        timestamppb.New(result.CreatedAt),
 			InUse:              true,
+			GeneratedAt:        timestamppb.New(result.CreatedAt),
 			GenerationDuration: convertDuration(result.Duration),
-			Sbom: &model.SBOMEntity_Cyclonedx{
-				Cyclonedx: convertBOM(bom),
-			},
 		}
+
+		if result.Error != nil {
+			sbom.Sbom = &model.SBOMEntity_Error{
+				Error: result.Error.Error(),
+			}
+			sbom.Status = model.SBOMStatus_FAILED
+		} else {
+			report, err := result.Report.ToCycloneDX()
+			if err != nil {
+				log.Errorf("Failed to extract SBOM from report: %s", err)
+				sbom.Sbom = &model.SBOMEntity_Error{
+					Error: err.Error(),
+				}
+				sbom.Status = model.SBOMStatus_FAILED
+			} else {
+				sbom.Sbom = &model.SBOMEntity_Cyclonedx{
+					Cyclonedx: convertBOM(report),
+				}
+			}
+		}
+		p.queue <- sbom
 	}()
 }
 
 func (p *processor) processImageSBOM(img *workloadmeta.ContainerImageMetadata) {
-	if img.SBOM == nil || img.SBOM.CycloneDXBOM == nil {
+	if img.SBOM == nil {
+		return
+	}
+
+	if img.SBOM.Status == workloadmeta.Success && img.SBOM.CycloneDXBOM == nil {
```

Comment:
can we have partial SBOM creation? for example CycloneDXBOM is provided but only a subpart of the analysers was successful. 

Also, IMO instead of returning direclty, we should still send a SBOM with an status error.

Reply:
I don't think so.